src/inmanta/server/services/userservice.py

```diff
@@ -0,0 +1,114 @@
+"""
+    Copyright 2023 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import logging
+
+import asyncpg
+
+import nacl.exceptions
+import nacl.pwhash
+from inmanta import const, data, protocol
+from inmanta.data import model
+from inmanta.protocol import common, exceptions
+from inmanta.server import SLICE_TRANSPORT, SLICE_USER
+from inmanta.server import protocol as server_protocol
+
+LOGGER = logging.getLogger(__name__)
+
+
+class UserService(server_protocol.ServerSlice):
```

Comment:
The user management endpoints that write to the database should raise an exception when they are invoked when authentication is disabled on the server. The error message should reference to the tool to setup the initial user.